docs/reference-guides/rancher-security/rancher-webhook-hardening.md

```diff
@@ -0,0 +1,133 @@
+```
+
+This is also the same config file where other admission plugins are configured, such as PodSecurity. If your distro or your setup uses additional admission plugins, configure those as well. For example, add [RKE2's PodSecurity configuration](https://docs.rke2.io/security/pod_security_standards) to this file.
+
+2. Create the kubeconfig file that the admission plugins refer to. Rancher Webhook only supports client certificate authentication, so generate a TLS key pair, and set the kubeconfig to use either `client-certificate` and `client-key` or `client-certificate-data` and `client-key-data`. For example:
+
+```
+# /etc/rancher/admission/kubeconfig
+apiVersion: v1
+kind: Config
+users:
+- name: 'rancher-webhook.cattle-system.svc'
+  user:
+    client-certificate: /path/to/client/cert
+    client-key: /path/to/client/key
+```
+
+3. Start the kube-apiserver binary with the flag `--admission-control-config-file` pointing to your AdmissionConfiguration file. The way to do this varies by distro, and isn't supported for some distros, such as hosted Kubernetes providers. Consult the documentation for your Kubernetes distribution.
+
+  For RKE2, `rke2-server` can be started with a config file like so:
+
+```
+# /etc/rancher/rke2/config.yaml
+kube-apiserver-arg:
+- admission-control-config-file=/etc/rancher/admission/admission.yaml
+kube-apiserver-extra-mount:
+- /etc/rancher/admission:/etc/rancher/admission:ro
+```
+
+:::warning
+Some distros set this flag by default. If your distro provisions its own AdmissionConfiguration, you must include it in your custom admission control config file. For example, RKE2 installs an AdmissionConfiguration file at `/etc/rancher/rke2/rke2-pss.yaml`, which configures the PodSecurity admission plugin. Setting `admission-control-config-file` in config.yaml will override this essential security setting. To include both plugins, consult [the Default Pod Security Standards documentation](https://docs.rke2.io/security/pod_security_standards) and copy the appropriate plugin configuration to your admission.yaml.
+:::
+
+4. If you're using Rancher to provision your cluster using existing nodes, create these files on the node before you provision them.
+
+  If you're using Rancher to provision your cluster on new nodes, allow the provisioning to complete, then use the provided SSH key and IP address to connect to the nodes, and place the RKE2 config file in the `/etc/rancher/rke2/config.yaml.d/` directory.
+
+5. After the cluster is configured with these credentials, configure the Rancher cluster agent to enable authentication in the webhook. Create a file containing these chart values:
+
+```
+# values.yaml
+auth:
+  clientCA: <base64-encoded certificate authority that signed the kube-apiserver's client certificates>
+  allowedCNs:
+  - <list of Common Names identifying the kube-apiserver's client certificates.>
+  - <if not provided, any cert signed by the given CA will be accepted.>
+```
```

Comment:
```suggestion
    ```yaml
    # values.yaml
    auth:
      clientCA: <base64-encoded certificate authority that signed the kube-apiserver's client certificates>
      allowedCNs:
      - <list of Common Names identifying the kube-apiserver's client certificates.>
      - <if not provided, any cert signed by the given CA will be accepted.>
    ```
```